src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/DesignTimeInputsBuildManagerBridge.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+
+using Microsoft.VisualStudio.ProjectSystem.VS.Automation;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.TempPE
+{
+    [Export(typeof(IDesignTimeInputsBuildManagerBridge))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasicLanguageService)]
+    internal class DesignTimeInputsBuildManagerBridge : UnconfiguredProjectHostBridge<IProjectVersionedValue<DesignTimeInputsDelta>, IProjectVersionedValue<DesignTimeInputsDelta>, IProjectVersionedValue<DesignTimeInputsDelta>>, IDesignTimeInputsBuildManagerBridge
+    {
+        private readonly IProjectThreadingService _threadingService;
+        private readonly IDesignTimeInputsChangeTracker _designTimeInputsChangeTracker;
+        private readonly IDesignTimeInputsCompiler _designTimeInputsCompiler;
+        private readonly VSBuildManager _buildManager;
+
+        /// <summary>
+        /// For unit testing purposes, to avoid having to mock all of CPS
+        /// </summary>
+        internal bool SkipInitialization { get; set; }
+
+        [ImportingConstructor]
+        public DesignTimeInputsBuildManagerBridge(IProjectThreadingService threadingService,
+                                                  IDesignTimeInputsChangeTracker designTimeInputsChangeTracker,
+                                                  IDesignTimeInputsCompiler designTimeInputsCompiler,
+                                                  VSBuildManager buildManager)
+             : base(threadingService.JoinableTaskContext)
+        {
+            _threadingService = threadingService;
+            _designTimeInputsChangeTracker = designTimeInputsChangeTracker;
+            _designTimeInputsCompiler = designTimeInputsCompiler;
+            _buildManager = buildManager;
+        }
+
+        /// <summary>
+        /// Get the list of design time monikers that need to have TempPE libraries created. Needs to be called on the UI thread.
+        /// </summary>
+        public string[] GetTempPEMonikers()
+        {
+            // Not using use the ThreadingService property because unit tests
+            _threadingService.VerifyOnUIThread();
+
+            Initialize();
+
+            return AppliedValue.Value.Inputs.ToArray();
+        }
+
+        /// <summary>
+        /// Gets the XML that describes a TempPE DLL, including building it if necessary
+        /// </summary>
+        public async Task<string> GetDesignTimeInputXmlAsync(string file)
+        {
+            // Not using use the ThreadingService property because unit tests
+            await _threadingService.SwitchToUIThread();
+
+            if (!SkipInitialization)
```

Comment:
Why can't you initialize this in tests?

Reply:
Just avoiding mocking heaps of CPS stuff that's used by the base class. I can if you think it's important.